source/Pawn.cpp

```diff
@@ -78,8 +123,24 @@ void Pawn::KeyboardEvensHandlers(Environment &env, Uint32 eventType, SDL_Keycode
 
 }
 
-Pawn::Pawn(const int x, const int y, const int width, const int height, const int color, const int speed)
-        : BaseObj(x, y, width, height, color, speed) {
+Direction Pawn::GetDirection() const
+{
+    return _direction;
 }
 
-Pawn::~Pawn() = default;
\ No newline at end of file
+void Pawn::SetDirection(const Direction direction)
+{
+    _direction = direction;
+}
+
+/* Old destroy
+void Pawn::Destroy(Environment& env) const
+{
+    const auto it = std::find(env.allPawns.begin(), env.allPawns.end(), this);
```

Comment:
this old code realy needed?

Reply:
Yea. After making all objects creation with a "new" I planning to move 121 to 127 lanes in main.cpp and uncomment this method.